lib/awesome_spawn.rb

```diff
@@ -123,8 +136,14 @@ def launch(env, command, spawn_options)
   def parse_command_options(command, options)
     options = options.dup
     params  = options.delete(:params)
-    env = options.delete(:env) || {}
+    env     = options.delete(:env) || {}
+    cmd_str = build_command_line(command, params)
+
+    if pipe = options.delete(:pipe)
+      pipe = Array(pipe)
+      cmd_str << " | " << parse_command_options(pipe[0], (pipe[1] || {}))[1]
```

Comment:
I've tried to keep all command line building in the CommandLineBuilder class, so, at a minumum this should live over there.  Also, not a fan of the `[0]` `[1]` stuff...would be better to find a way to make those named variables for readability...i.e. instead of `pipe = Array(pipe)`, prefer `cmd, params = Array(pipe)`

Reply:
> I've tried to keep all command line building in the CommandLineBuilder class,

@Fryguy So kind of the main reason I avoided that a didn't want to pollute that class with the `:pipe` option, since there might be some command out there that takes `--pipe` as a flag, and it seemed like I would need to do some heavy duty surgery to handle that properly in that class.

That said, not opposed to doing it that way, but that is the main rational (at least from what I remember doing this a few weeks ago) for why I didn't use that class to do this.

> Also, not a fan of the `[0]` `[1]` stuff...

That is fair.  I will look in doing that, as your suggestion looks much better.  I also have a unreasonable urge to keep lines to 80 chars, so that was part of the madness that probably led to the `[0]` and `[1]` stuffs.

Reply:
If "--pipe" were a flag it would have to live inside of the :params key anyway.  Putting the :pipe key up near :env and :params would be more of a potential conflict with Open3#capture3, since we passthrough every other option at that level to capture3.  We've already taken that stance with `:params` and we've been ok, so I'm ok with also taking that stance with `:pipe`.  If Open3#capture3 adds its own pipe option, then we can deal with it then.